### PR TITLE
Removing stub from isotopic depletion init file

### LINKED ISSUE #1458: Remove unnecessary `isotopicDepletion` stub
This file is not useful. If a user wants to use the `STACK_ORDER.DEPLETION` constant, they can just use it directly from the `interfaces` module. None of the other imports in this file are used.

https://github.com/terrapower/armi/blob/9da65879499bda3b9f1e9829e3d9f935275eeaa7/armi/physics/neutronics/isotopicDepletion/__init__.py#L15-L23

Unfortunately we have an internal plugin that directly imports `ORDER` from here. So we can't just delete this. But it should be slated for removal.

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Removing some unnecessary stub code from the isotopic depletion `__init__.py` module-identification file.

## Why is the change being made?

To close #1458

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/physics/neutronics/isotopicDepletion/__init__.py [hunk 1]
```diff
@@ -13,11 +13,3 @@
 # limitations under the License.
 
 """The depletion physics package contains utility/framework code related to the physics of transmutation and decay."""
-# ruff: noqa: F401
```
Comment (keckler) [inline armi/physics/neutronics/isotopicDepletion/__init__.py:16]: Is there a good reason we don't just remove this file entirely? `__init__.py` files are not required beyond Python 3.3, and I don't see any imports breaking if we remove this.

Side point is that we don't even mention this `isotopicDepletion` namespace in this discussion:
https://github.com/terrapower/armi/blob/827d223cd6fc085c039390a2c3a0391f4e611fe5/armi/physics/__init__.py#L15-L46

Comment (john-science) [reply armi/physics/neutronics/isotopicDepletion/__init__.py:16]: > Is there a good reason we don't just remove this file entirely? __init__.py

@keckler @albeanth Sorry, was this the confusion originally?

In Python, you MUST have a file called `__init__.py` in every single folder that you want to put Python files in.  This is part of the language, and how the Python interpreter knows to include "folders of Python files" in importable libraries.  (This requirement has softened a bit in recent years, but is still firm in some scenarios.)

As long as there are other Python files in this folder, the `__init__.py` files exists.

Comment (keckler) [reply armi/physics/neutronics/isotopicDepletion/__init__.py:16]: This is not true. See, for instance: https://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html#the-init-py-trap

By including that `__init__.py` file, we are preventing the use of namespace packages at that level. Perhaps that IS something we want to do, but that would come as a surprise to me, and it is also a bit counter to some text we have in the codebase here: https://github.com/terrapower/armi/blob/827d223cd6fc085c039390a2c3a0391f4e611fe5/armi/physics/__init__.py#L48

Comment (john-science) [reply armi/physics/neutronics/isotopicDepletion/__init__.py:16]: Okay, I tested it. Deleted.

**Comment by john-science** [comment]
I have tested this thoroughly downstream, and found no issues of any kind.

**Pull request merged**